conda_oci_mirror/tests/conftest.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+
+import pytest
+from xprocess import ProcessStarter
+
+from conda_oci_mirror.oras import oras  # noqa
+
+# The setup.cfg doesn't install the main module proper
+here = os.path.dirname(os.path.abspath(__file__))
+root = os.path.dirname(os.path.dirname(here))
+sys.path.insert(0, root)
+sys.path.insert(0, here)
+
+import conda_oci_mirror.defaults as defaults  # noqa
+
+# import conda_oci_mirror.defaults as defaults
+from conda_oci_mirror.mirror import Mirror  # noqa
+
+
+def check_media_type(layer):
+    """
+    Ensure the layer media type matches what we expect
+    """
+    if layer["path"].endswith("repodata.json"):
+        assert layer["media_type"] == defaults.repodata_media_type_v1
+    elif layer["path"].endswith("conda"):
+        assert layer["media_type"] == defaults.package_conda_media_type
+    elif layer["path"].endswith("bz2"):
+        assert layer["media_type"] == defaults.package_tarbz2_media_type
+    elif layer["path"].endswith("info.tar.gz"):
+        assert layer["media_type"] == defaults.info_archive_media_type
+    elif layer["path"].endswith("index.json"):
+        assert layer["media_type"] == defaults.info_index_media_type
+    else:
+        raise ValueError(f"Unexpected layer content type {layer}")
+
+
+def registry_host():
+    return os.environ.get("registry_host") or "http://127.0.0.1"
+
+
+def registry_port():
+    return os.environ.get("registry_port") or 5000
+
+
+def registry_url():
+    return f"{registry_host()}:{registry_port()}"
+
+
+def test_user():
+    return "dinosaur"
+
+
+def delete_tags(registry, channel, subdir, name):
+    """
+    Delete all tags for a repo
+    """
+    tags = f"{registry_url()}/{test_user()}/{channel}/{subdir}/{name}"
+    try:
+        tags = oras.get_tags(tags)
+        print(f"Got tags to delete {tags}")
+    except Exception:
+        return
+
+    for tag in tags:
+        # get digest of manifest
+        print(f"Deleting tag {tag} for {name}")
+        manifest_url = f"{registry_url()}/v2/{test_user()}/{channel}/{subdir}/{name}/manifests/{tag}"
+        response = oras.do_request(
+            manifest_url,
+            "HEAD",
+            headers={"Accept": "application/vnd.oci.image.manifest.v1+json"},
+        )
+        digest = response.headers.get("Docker-Content-Digest")
+        delete_url = (
+            f"{registry_url()}/v2/dinosaur/{channel}/{subdir}/{name}/manifests/{digest}"
+        )
+        response = oras.do_request(delete_url, "DELETE")
+
+        if response.status_code != 202:
+            print(response.text)
+            raise RuntimeError("Failed to delete tag")
+
+
+@pytest.fixture
+def oci_registry(xprocess):
+    class Starter(ProcessStarter):
```

Comment:
Oh this is super cool! https://pytest-xprocess.readthedocs.io/en/latest/ I've never looked at this library before! 

So this handles bringing the registry up/down for the specific test (in a process?) How does that relative to the GitHub action service? Do we need both of them?

Reply:
okay - working on adding delete function to oras-py, and I'm also going to add the retry there. I realize I made an issue for it last year (and then promptly forgot about it, it seems!)

Reply:
Update: finished that, going to try updating the PR here with these functions https://github.com/oras-project/oras-py/pull/73

Reply:
unfortunately we don't really need the function anymore since we remove / restart the entire registry now :D But I love that it made it's way into oras-py. Functions to `delete_blob` would be a great addition as well, btw! :)

Reply:
And yes, xprocess starts a subprocess and checks if the service is up by running a regex against the output (`pattern = r".*listening on \[::\]:5000.*"`). Once it sees the pattern in the output, the tests can be run.

Reply:
> Functions to delete_blob would be a great addition as well, btw! :)

Then it shall be done!  https://github.com/oras-project/oras-py/issues/74

> And yes, xprocess starts a subprocess and checks if the service is up by running a regex against the output (pattern = r".*listening on \[::\]:5000.*"). Once it sees the pattern in the output, the tests can be run.

Love that. I have a bunch of projects that would warrant this refactor, so I'll do them slowly over time. It's like a piece to a puzzle I was missing!